### ISSUE TITLE
Allow to set DesktopFileName via cli flag

### ISSUE THREAD

**Issue by helmecke** [qutebrowser/qutebrowser#5245]
Allow to set DesktopFileName through cli flag. To make it easier to match specific qutebrowser instances to i3wm/swaywm workspaces.

https://github.com/qutebrowser/qutebrowser/blob/master/qutebrowser/app.py#L95

**Comment by helmecke** [comment]
I see, it was the only value I found which correspondes to the app_id in sway/wayland. Setting `instance` doesn't work on Wayland. Is there a way to set the app_id?

**Comment by The-Compiler** [comment]
I stand corrected - after talking to Drew Devault in IRC, it looks like `app_id` is only tangientally related to the desktop filename, and the Qt thing is a bit of a misnomer. I'll need to check if there are any other places where that information is used, though.

**Comment by markstos** [comment]
I tested changing the DesktopFileName and re-launching `qutebrowser` with a temp basedir:

https://github.com/qutebrowser/qutebrowser/blob/d397725ebfbf375c17c1ddd20896e84454bb1a0b/qutebrowser/app.py#L97

It worked-- The resulting "app_id" was changed under Wayland.

It also wouldn't see it as a deal breaker if DesktopFileName _does_ need to point a .desktop file. That's exactly how Chromium implements SSBs / Single Site Browsers / PWAs-- a new ".desktop" file is created in `~/.local/share/applications` that launches Chromium with a different profile and icon.

If `qutebrowser` can handle the other parts, I'm happy to write my own `.desktop` files for alternate instances. Creating those could also be a feature added to a companion project like `qutebrowser-profile`. https://github.com/jtyers/qutebrowser-profile
